internal/yggdrasil/yggdrasil.go

```diff
@@ -43,15 +44,18 @@ type Handler struct {
 	claimer              *storage.Claimer
 	initialNamespace     string
 	recorder             record.EventRecorder
+	repositoryAuth       images.RepositoryAuthAPI
 }
 
-func NewYggdrasilHandler(deviceRepository edgedevice.Repository, deploymentRepository edgedeployment.Repository, claimer *storage.Claimer, initialNamespace string, recorder record.EventRecorder) *Handler {
+func NewYggdrasilHandler(deviceRepository edgedevice.Repository, deploymentRepository edgedeployment.Repository,
+	claimer *storage.Claimer, initialNamespace string, recorder record.EventRecorder, repositoryAuth images.RepositoryAuthAPI) *Handler {
 	return &Handler{
 		deviceRepository:     deviceRepository,
 		deploymentRepository: deploymentRepository,
 		claimer:              claimer,
 		initialNamespace:     initialNamespace,
 		recorder:             recorder,
+		repositoryAuth:       repositoryAuth,
```

Comment:
a question about the naming, in terms of symmetry with `deviceRepository` and `deploymentRepository`, shouldn't it be called `AuthRepository` ?

Reply:
If kept the naming in line, that would be RepositoryAuthRepository; It's not a repository for auth, but (image) respository authentication.